PRC History Review.js

```diff
@@ -0,0 +1,297 @@
+				}
+				else {
+					// if it is homepage link, parse the one-link raw title into title and other info.
+					const rawTitle = linkEl.textContent;
+					const rawTitleParts = rawTitle.split(':');
+					const firstPart = rawTitleParts[0];
+
+					const hasSeriesInfo = firstPart && firstPart.toLowerCase().endsWith('series');
+					if (hasSeriesInfo) {
+						newItem.seriesTitle = firstPart;
+						// remove series title, leaving the rest as title;
+						rawTitleParts.shift();
+						newItem.title = rawTitleParts.join(":");
+					}
+					else {
+						newItem.title = rawTitle;
+					}
+				}
+
+				// add issue info
+				if (isIssue) {
+					const issueInfoEl = doc.querySelectorAll('h5')[0];
+					if (issueInfoEl) {
+						const issueInfoArray = issueInfoEl.textContent.split('★');
+						if (issueInfoArray.length === 3) {
+							const volume = issueInfoArray[0];
+							newItem.volume = volume.toLowerCase().replace('volume', '').trim();
+							const issue = issueInfoArray[1];
+							newItem.issue = issue.toLowerCase().replace('number', '').trim();
+							const date = issueInfoArray[2];
+							newItem.date = ZU.strToISO(date);
+						}
+					}
+				}
+				else if (isSingleArticle) {
+					// if it's single article page, the issue info is two divs above the link div.
+					const issueInfoElIndex = index - 2;
+					if (issueInfoElIndex >= 0) {
+						const issueInfoEl = doc.querySelectorAll('h5')[issueInfoElIndex];
+						if (issueInfoEl) {
+							const issueInfoArray = issueInfoEl.textContent.split('★');
+							if (issueInfoArray.length === 3) {
+								const volume = issueInfoArray[0];
+								newItem.volume = volume.toLowerCase().replace('volume', '').trim();
+								const issue = issueInfoArray[1];
+								newItem.issue = issue.toLowerCase().replace('number', '').trim();
+								const date = issueInfoArray[2];
+								newItem.date = ZU.strToISO(date);
+							}
+						}
+					}
+				}
+				else {
+					// if it's homepage article, the issue info is in the previous div.
+					const lastIndex = index - 1;
+					if (lastIndex >= 0) {
+						const issueInfoEl = articleLinkEls[lastIndex];
+						const issueInfoArray = issueInfoEl.textContent.split('★');
+						if (issueInfoArray.length === 2) {
+							const issue = issueInfoArray[0];
+							newItem.issue = issue.toLowerCase().replace('number', '').trim();
+							const date = issueInfoArray[1];
+							newItem.date = ZU.strToISO(date);
+						}
+					}
+				}
+
+				// add author
+				if (isIssue) {
+					const authorEl = articleLinkEls[index + 1];
+					if (authorEl) {
+						// in case there are multiple authors
+						const authorTexts = authorEl.textContent.trim().split(' and ');
+						authorTexts.forEach((authorText) => {
+							const authorName = authorText.split(',')[0];
+							if (authorName) {
+								newItem.creators.push(ZU.cleanAuthor(authorName, 'author', false));
+							}
+						});
+					}
+				}
+
+				if (isSingleArticle) {
```

Comment:
Can these `if`s be condensed, or could this single `scrape` function be split into multiple? The logic here is difficult to follow and the calls to `scrape` with two unlabeled boolean arguments are a bit opaque.

Reply:
Thanks! I tried but since every Zotero field needs to change according to these variables, it might be easier doing with one function. I did remove the unlabled variables and simplifyy the logics.